main.tf

```diff
@@ -16,16 +16,19 @@ module "project_factory_project_services" {
     "cloudkms.googleapis.com",          // KMS
     "compute.googleapis.com",           // required for datadog monitoring
     "cloudasset.googleapis.com",        // required for datadog monitoring
-    "secretmanager.googleapis.com"      // required for secrets
+    "secretmanager.googleapis.com",
+    "cloudresourcemanager.googleapis.com" // required for secrets
   ]
 }
 
 locals {
-  fqdn           = var.subdomain == null ? var.domain_name : "${var.subdomain}.${var.domain_name}"
-  url_prefix     = var.ssl ? "https" : "http"
-  url            = "${local.url_prefix}://${local.fqdn}"
-  create_bucket  = var.bucket_name == ""
-  create_network = var.network == null
+  fqdn            = var.subdomain == null ? var.domain_name : "${var.subdomain}.${var.domain_name}"
+  url_prefix      = var.ssl ? "https" : "http"
+  url             = "${local.url_prefix}://${local.fqdn}"
+  create_bucket   = var.bucket_name == ""
+  create_network  = var.network == null
+  bucket_location = var.bucket_location == "" ? "US" : var.bucket_location
+  database_region = var.database_region == "" ? "us-central1" : var.database_region
```

Comment:
Hey @amanpruthi, I attempted this going from `v4.0.0` to this branch, and setting the region like this and changing the key to `null` will both cause the database to be *recreated*. 

We can't let this happen, or it will cause a lot of problems.

![Screen Shot 2024-07-16 at 10 01 07 AM](https://github.com/user-attachments/assets/057590e8-39fd-4812-84f2-6248cb86e0a1)

Reply:
I think we want to default to `var.region` if `var.database_region` isn't set

Reply:
We have added the suggested changes for database region, And the initial screenshot attached ( Service Account) not related to this branch.